xbmc/URL.cpp

```diff
@@ -355,29 +355,29 @@ void CURL::SetOptions(const std::string& strOptions)
        strOptions[0] == ';' ||
        strOptions.find("xml") != std::string::npos)
     {
-      m_strOptions = strOptions;
+      m_strOptions = std::move(strOptions);
```

Comment:
for example here
before with `const std::string&` data arg was not copied, but copied only to the m_strOptions

now you are creating and copying the data to a new string in the argument
you do the move after

in this way you always allocate new memory, also when the code dont reach the line 358,
this is not an improvement at all...

Reply:
in other hand, the "move" improvement is nullified by the creation/copy that now happens in the method definition since its anymore const referenced so no improvements here and similar other CURL method modified

Reply:
Okay, lets talk numbers then: https://quick-bench.com/q/ZbsYKqaKkyRSOCqU0AZwf2mo-WQ

`std::string_view` is only faster in the most trivial case. Maybe the measurements are flawed but in my head it makes sense.

Reply:
thanks for the example, to be sure i replied with godbolt https://godbolt.org/z/a4naG61nh mine so elegant but not too much time to spend, i can confirm that yes, for the speed look like most the time "move" give better performance,
despite by using "move" you allocate one more time on the method define,
in definitive i think this way can be good for small data strings, but not for strings with big data to avoid new mem allocation, so i discovered new things

Reply:
We are still not on the same page :wink: Here is a cleaned up version of the benchmark and we can see that the string size isn't a big factor: https://quick-bench.com/q/FKJI4T51sncva8KTbAlYA5ttVLg

In your Godbolt link as well `std::string` is faster with `-O3`. But that test does to many optimizations anyway, the `DoNotOptimize` inserts in the benchmark are there for good reason :slightly_smiling_face: 

> despite by using "move" you allocate one more time on the method define

That's not correct. Comparing the functions:
```c++
void SetStringView(std::string_view sv /*Never allocates*/) {
  if(sv.length() > 0) {
    m_string = sv; // Always allocates if not SSO
  }
}
```
```c++
void SetString(std::string s /*Only allocates if not moved into and not SSO*/) {
  if(s.length() > 0) {
    m_string = std::move(s); // Never allocates
  }
}
```
So with `std::string_view` there is always one guaranteed allocation (if no SSO) while with `std::string` this allocation can be avoided as well if a string is moved into the function. If not there is also only one allocation.